File: tools/bench/answers/python_haiku/s2/11_collatz_max.py
def collatz_steps(n):
    steps = 0
    while n != 1:
        if n % 2 == 0:
            n //= 2
        else:
            n = 3 * n + 1
        steps += 1
    return steps

max_n = 1
max_steps = 0
for n in range(1, 11):
    steps = collatz_steps(n)
    if steps > max_steps:
        max_steps = steps
        max_n = n
print(f"{max_n} {max_steps}")

File: tools/bench/answers/go_sonnet/s2/11_collatz_max.go
package main

import "fmt"

func collatzSteps(n int) int {
	steps := 0
	for n != 1 {
		if n%2 == 0 {
			n /= 2
		} else {
			n = 3*n + 1
		}
		steps++
	}
	return steps
}

func main() {
	maxN, maxSteps := 0, -1
	for n := 1; n <= 10; n++ {
		s := collatzSteps(n)
		if s > maxSteps {
			maxSteps = s
			maxN = n
		}
	}
	fmt.Println(maxN, maxSteps)
}
